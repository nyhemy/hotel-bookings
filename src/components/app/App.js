import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import styles from './App.module.css';
import Nav from '../nav/Nav';
import Login from '../login/Login';
import Logout from '../logout/Logout';
import Reservations from '../reservations/Reservations';
import Rooms from '../rooms/Rooms';

/**
 * 
 * @param {*} props 
 */
const App = (props) => {

  const [loggedIn, setLoggedIn] = useState(false);
  const [isManager, setIsManager] = useState(false);
  
  return (
    <div>
    <Router>
      <Nav loggedIn={loggedIn} isManager={isManager}/>

      <Switch>
        <Route exact path="/" render={() => <Login login={() => setLoggedIn(true)} loggedIn={loggedIn} isManager={isManager} />}/>
        <Route exact path="/reservations" component={Reservations} />
        <Route exact path="/room-types" component={Rooms} />
        <Route exact path="/logout" render={() => <Logout logout={() => setLoggedIn(false)} />} />
      </Switch>

    </Router>
    </div>
  );
}

export default App;
