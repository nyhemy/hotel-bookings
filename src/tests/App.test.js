import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import App from '../components/app/App';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

test('renders w/o crashing', () => {
  const div = document.createElement('div');
  render(<App />, div);
  unmountComponentAtNode(div);
});

test('app component matches snapshot', () => {

  const component = renderer.create(<App />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

});