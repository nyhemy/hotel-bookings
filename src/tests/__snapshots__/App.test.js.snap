// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component matches snapshot 1`] = `
Array [
  <nav
    className="nav"
  >
    <h3>
      Hotel Bookings
    </h3>
  </nav>,
  <div
    className="container"
  >
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="notification"
      >
        
      </div>
      <br />
      <div>
        <input
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
        />
      </div>
      <div>
        <input
          name="password"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
      </div>
      <button
        className="button"
        type="submit"
      >
        login
      </button>
    </form>
  </div>,
]
`;
